### DISCUSSION TITLE
NoCode Table Schema Recommendation

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#16073] [Feature Requests]
People with no or few knowledge about databases and normalization can be supported. I propose a system to guide these people at the creation of database table schemas.

my idea is that a developer will give the system a list of variables he wants to use (with or without datatype) and the system then suggests them a database Schema. This way a database/backend could be easier setup.

i since I haven’t found any tool offering this, I will maybe write a paper and a small tool for this.

maybe this idea/tool could then be implemented in Directus. Any thought about this?

**Comment by rijkvanzanten** [comment]
Curious to see how we could correctly suggest systems based on a free input 🤔

**Comment by NilsBaumgartner1994** [comment]
One idea would be machine learning (ai) or what my base idea is, to scrape through open source projects. From these open source projects collect all table and fill a database with fields and relations. 
This would be a part of my paper „data clump refactoring“. With these informations and the users input a nice algorithm would then suggest the Schema for a user.


Beste Grüße,
Nils Baumgartner

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
